accepted/2025/Multiple-Equivalent-Framework-Support-TFM-As-Aliases.md

```diff
@@ -0,0 +1,463 @@
+- dotnet package add, dotnet package update and dotnet package list will all be updated to support per framework operations in case they do not support it already.
+
+### Pack and duplicate frameworks
+
+There is no intuitive solution to the pack scenario.
+Given a project with multiple versions of the same framework, pack will fail by default.
+
+NuGet will hinge off off IncludeBuildOutput, IncludeContentInPack and SuppressDependenciesWhenPacking to decide to skip packing for a framework, effectively allowing a framework to be completely excluded.
+
+Pack can only succeed when one of the 2 frameworks doesn't have both of the following `IncludeContentInPack == true or IncludeBuildOutput == true or SuppressDependenciesWhenPackaging == false`.
+
+The pack experience likely requires some learning which may not be possible until we see more scenarios utilizing duplicate frameworks.
+
+> Note that as part of this proposal, there may be some scenarios where alias usage in per framework commands will be validated and potentially fixed. This will discovered during the implementation and won't be contained in detail in the current design.
+
+### Technical explanation
+
+There are many changes required to implement this feature.
+
+1. [Restore output (assets file) changes](#restore-output-assets-file-changes)
+1. [Components iterating target frameworks](#components-iterating-target-frameworks)
+1. [Lock file changes](#lock-file-changes)
+1. [ProjectReference changes](#project-to-project-references)
+1. [Visual Studio Challenges](#visual-studio-challenges)
+
+The changes listed are largely what NuGet needs to implement in code we own, but there will be impacts to other components as well.
+In particular the .NET SDK may need some changes to be able to use the assets file correctly.
+Majority of this had fortunately been done in past iterations and the .NET SDK currently depends on are NuGet APIs.
+Other tools, like various Visual Studio components, might have assumptions about TFM uniqueness in the project, which this feature changes.
+Therefore, while the NuGet team works very closely with the .NET SDK team and we will ensure that building projects work, there will increased risk of other features and tools breaking, which would normally not be affected by NuGet changes.
+
+### Restore output (assets file) changes
+
+In PackageReference, NuGet has a contract with the .NET SDK where NuGet writes the assets file and the .NET SDK reads it.
+The .NET SDK uses NuGet libraries to parse this file, and as such, as long as NuGet's APIs don't change, the .NET SDK will not need any code changes.
+PackageReference is also supported by non-SDK style projects, which use [dotnet/NuGet.BuildTools](https://github.com/NuGet.BuildTools) which, despite the name, is owned by the non-SDK project system team, not NuGet.
```

Comment:
<https://github.com/NuGet.BuildTools> gives me 406 Not Acceptable.  Is that the correct link?

Reply:
<https://github.com/dotnet/NuGet.BuildTasks> exists; did you mean that?

Reply:
> https://github.com/dotnet/NuGet.BuildTasks exists; did you mean that?

That's exactly what I meant. 
I legit don't know how I ended up with that link haha. 

Thanks!

Reply:
Fixed now.